### PR TITLE
Rename SlopeLimiters -> Limiters

### PR DESCRIPTION
**Pull request by fmahebert**
## Proposed changes

Now that many of our limiters are not slope limiters, it is time to rename SlopeLimiters -> Limiters.

This is partial progress towards resolving #1442 

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodType.cpp
Comment (nilsdeppe) [inline src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodType.cpp:13]: An FYI can be done later, the stream operator should be in the `Limiters` namespace for ADL to work correctly. As it stands, there will be situations where the stream operator won't be found because only the `std` and `Limiters` namespaces will be searched. I'll leave it up to you to decide whether to deal with this now or when/if it happens :)

**Review by nilsdeppe — DISMISSED** [review]
LGTM! Thanks for splitting the commits the way you did. Made it super easy to review! :D 

I have one comment that I'd just like you to respond to whether you want to do this now or wait until it's an issue :)

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodType.cpp
Comment (fmahebert) [reply src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodType.cpp:13]: Ah, right. I'll do this in a subsequent PR once this one is merged :)

**Comment by fmahebert** [comment]
I noticed I had two renaming changes in the wrong commits, likely from incorrectly resolving a conflict during a rebase. I've force-pushed an update to clean those up.

**Review by nilsdeppe — APPROVED** [review]
Okay :+1:

**Review by kidder — APPROVED** [review]

**Pull request merged**
